acq4/util/future.py

```diff
@@ -185,9 +185,9 @@ def wait(self, timeout=None, updates=False, pollInterval=0.1):
             else:
                 msg = f"Task {self} did not complete: {err}"
             if self._excInfo is not None:
-                raise RuntimeError(msg) from self._excInfo[1]
+                raise self._excInfo[1] from self._excInfo[1]
             else:
-                raise RuntimeError(msg)
+                raise self.StopRequested(msg)
```

Comment:
Well let'sthink about it .. the whole reason for StopRequested is that it gives the owner of this future the option to stop itself as well. Is that what we want to happen here? (My feeling is no)

Reply:
Hrm. For that scenario, the user just said "stop doing this task", and if the owner of that task is also capable of being told to stop, then yeah, it probably should stop as well. You think no? That'll often look like `_future.waitFor(other_future)`, and that's a clear yes. If it's instead something like `if other_future.wasInterrupted(): ...some recovery... then raise via: other_future.wait()`, that's a less clear relationship. Maybe `other_future` is a subtask that can be canceled but then the parent task shouldn't consider itself to have been canceled, but it still wants to blow itself up with the subtask's exception, without adding any additional context as to why it shouldn't be considered a canceled task... That seems weird. With this change in place, one can still raise a different exception from the parent task, which would importantly offer the additional context as to _why_ it's not a "canceled task". It would lose the subtask's stack, in that case.

Reply:
All that said, semantically, StopRequested is not correct. It's conflating "please stop now" and "this future was stopped by user request", which is what you were getting at. It's just that the 2 different contexts - inside a task or monitoring a task from the gui - are distinct enough that this confusion won't be much of a problem.